### ISSUE TITLE
triggers-webhook-cert expired after a year

### ISSUE THREAD

**Issue by robertlestak** [tektoncd/triggers#1161]
# Expected Behavior

- Tekton Triggers to auto-renew the `triggers-webhook-cert` that is auto-provisioned at install time 

# Actual Behavior

- `triggers-webhook-cert` expires after one year

# Steps to Reproduce the Problem

1. Install `tekton-triggers`
2. Wait a year
3. After a year, try to apply a manifest to triggers, and you will receive the following error: `x509: certificate has expired or is not yet valid`

# Additional Info

- Kubernetes version:

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:11:29Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.15-eks-ad4801", GitCommit:"ad4801fd44fe0f125c8d13f1b1d4827e8884476d", GitTreeState:"clean", BuildDate:"2020-10-20T23:27:12Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version: `v0.15.1`

**Comment by dibyom** [comment]
Hmm...we rely on the knative/pkg certificates controller for renewing/changing the secrets at the right time. Have you seen the same issue happen with your pipelines install too or no? If not, one possibility is that the certificates controller tried to update the cert but could not due to some reason (missing RBAC roles perhaps?)

Do you happen to have logs for the `tekton-triggers-webhook` deployment in the `tekton-pipelines` namespace?

**Comment by dibyom** [comment]
Thanks for the logs and sorry for the delay in responding. I don't see anything suspicious there (other than the bad certificate errors)

Two questions:
1. How did you fix the issue? Did you manually roll the certs? 
2. Is it possible to get the RBAC roles for the service account that the webhook pod is running as? Specifically, does it have permissions to get/list/update secrets?

One thing I did notice - Triggers version is 0.4 which is fairly old. There is a possibility that there was a bug in the certificates controller from knative/pkg which has since then been fixed. So updating to a new Triggers version might help.

**Comment by robertlestak** [comment]
No worries, thanks for digging into this. 

I do agree that the version is quite old - I'll take this as the driver to upgrade the environment, and come back here in a year if the issue crops back up :)

**Comment by freefood89** [comment]
@robertlestak thanks for the tip!

For some reason when I recreated the secret and restarted the webhook/controller pod they didn't populate the secret with a fresh cert. I figured knative would create it, but maybe I'm not understanding how it works

On a side note, another workaround for forcing your Trigger to go deploy is to delete the `ValidatingWebhookConfiguration` and `MutatingWebhookConfiguration` assuming that your Trigger is valid and doesn't need default values set.
